kythe/typescript/indexer.ts

```diff
@@ -496,39 +565,29 @@ class StandardIndexerContext implements IndexerHost {
    * optionally specified to help disambiguate nodes with multiple declarations.
    * See the documentation of Context for more information.
    */
-  getSymbolName(sym: ts.Symbol, ns: TSNamespace, context?: Context): VName {
-    let vnames = this.symbolNames.get(sym);
-    let declarations = sym.declarations;
-
-    // Symbols with multiple declarations are disambiguated by the context
-    // they are used in.
-    const contextApplies = context !== undefined && declarations.length > 1;
+  getSymbolName(
+      sym: ts.Symbol, ns: TSNamespace, context: Context = Context.Any): VName {
+    const stored = this.symbolNames.get(sym, ns, context);
+    if (stored) return stored;
 
-    if (!contextApplies && vnames && vnames[ns]) return vnames[ns]!;
-    // TODO: update symbolNames table to account for context kind
-
-    if (!declarations || declarations.length < 1) {
+    let declarations = sym.declarations;
+    if (declarations.length < 1) {
       throw new Error('TODO: symbol has no declarations?');
     }
 
-    // Disambiguate symbols with multiple declarations using a context. This
-    // only applies to getters and setters currently.
-    if (contextApplies) {
+    // Disambiguate symbols with multiple declarations using a context.
+    if (sym.declarations.length > 1) {
       switch (context) {
         case Context.Getter:
           declarations = declarations.filter(ts.isGetAccessor);
           break;
         case Context.Setter:
           declarations = declarations.filter(ts.isSetAccessor);
           break;
+        default:
+          break;
       }
     }
-    // Otherwise, if there are multiple declarations but no context is
-    // provided, try to return the getter declaration.
-    else if (declarations.length > 1) {
-      const getDecls = declarations.filter(ts.isGetAccessor);
-      if (getDecls.length > 0) declarations = getDecls;
-    }
 
     const decl = declarations[0];
```

Comment:
Maybe comment that you are taking the first decl, and so if the Context is Any, it will be "lazily" decided.
Then again, that's been mentioned in at least two other places, so this is unnecessary.

Reply:
I think it's okay to have the documentation where it already is. Not adding here unless big opposition.